hkmc2/shared/src/main/scala/hkmc2/codegen/StackSafeTransform.scala

```diff
@@ -120,10 +151,11 @@ class StackSafeTransform(depthLimit: Int)(using State):
     defn match
     case d: FunDefn => rewriteFn(d)
     case _: ValDefn => defn
-    case ClsLikeDefn(sym, k, parentPath, methods, privateFields, publicFields, preCtor, ctor) =>
+    case ClsLikeDefn(owner, isym, sym, k, paramsOpt, 
+      parentPath, methods, privateFields, publicFields, preCtor, ctor) =>
       ClsLikeDefn(
-        sym, k, parentPath, methods.map(rewriteFn), privateFields, publicFields,
-        rewriteBlk(preCtor), rewriteBlk(ctor) // TODO: do we need to rewrite the preCtor?
```

Comment:
> // TODO: do we need to rewrite the preCtor?

So I assume the answer was yes? Or was the comment lost in the merge?

Reply:
Assuming the preCtor can have arbitrary code, yes